src/Microsoft.AspNetCore.Razor.Tools/ServerCommand.cs

```diff
@@ -34,37 +34,55 @@ protected override bool ValidateArguments()
         protected override Task<int> ExecuteCoreAsync()
         {
             // Make sure there's only one server with the same identity at a time.
-            using (var mutex = new Mutex(initiallyOwned: true, name: MutexName.GetServerMutexName(Pipe.Value()), createdNew: out var holdsMutex))
+            var serverMutexName = MutexName.GetServerMutexName(Pipe.Value());
+            Mutex serverMutex = null;
+            var holdsMutex = false;
+
+            try
             {
-                if (!holdsMutex)
-                {
-                    // Another server is running, just exit.
-                    Error.Write("Another server already running...");
-                    return Task.FromResult(1);
-                }
+                serverMutex = new Mutex(initiallyOwned: true, name: serverMutexName, createdNew: out holdsMutex);
+            }
+            catch (Exception ex)
+            {
+                // The Mutex constructor can throw in certain cases. One specific example is docker containers
+                // where the /tmp directory is restricted. In those cases there is no reliable way to execute
+                // the server and we need to fall back to the command line.
+                // Example: https://github.com/dotnet/roslyn/issues/24124
+
+                Error.Write($"Server mutex creation failed. {ex.Message}");
+
+                return Task.FromResult(1);
+            }
+
+            if (!holdsMutex)
```

Comment:
It looks like the mutex can leak right?

Reply:
Maybe I misunderstand this but wouldn't calling serverMutex.Dispose() here prematurely release the mutex held my the server already running?